build/wrapper.sh

```diff
@@ -62,6 +62,34 @@ if [ -z "$TRAP" ]; then
     exit
 fi
 
+# See if we want to just run a script from the bundle
+if [ "$1" = '-X' -a -z "$2" ]; then
```

Comment:
Put all this under the same `if [ "$1" = '-X' ];` block. i.e. move `[ -z "$2"]` to the if block below, you could even merge `-z "$2"` and `-f "$script"` and output something that makes sense in both cases.